dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,11 +92,14 @@ public void join(WorkerInfo workerInfo) throws IOException {
     byte[] serializedEntity = entity.serialize();
     // If there's existing entry, check if it's me.
     if (existingEntityBytes != null) {
-      // It's not me, something is wrong.
+      // It's not me, or not the same me.
       if (!Arrays.equals(existingEntityBytes, serializedEntity)) {
```

Comment:
why compare by the serialized form? The `WorkerServiceEntity` implements `equals` and ignores comparing `WorkerNetAddress` deliberately. If compared by the serialized form, the netaddress would be taken into account, which I find confusing.

Reply:
in bare metal cases, if a worker restart none of the fields in workerserviceentity will get changed, and the content should be the same, so we don't need to insert the kv pair again to etcd. if not , then thats a case of the worker changing hostname like in k8s, hence comparing the serialized form, just to say is it the same me on etcd? or its a different me, and im about to overwriting the content